fe/fe-core/src/main/java/com/starrocks/connector/iceberg/IcebergMetadata.java

```diff
@@ -254,14 +260,33 @@ private List<RemoteFileInfo> getRemoteFileInfos(IcebergTable table, long snapsho
                 scan = scan.filter(icebergPredicate);
             }
 
-            CloseableIterable<CombinedScanTask> combinedScanTasks = scan.planTasks();
-            for (CombinedScanTask combinedScanTask : combinedScanTasks) {
-                for (FileScanTask fileScanTask : combinedScanTask.files()) {
-                    builder.add(fileScanTask);
+            CloseableIterable<FileScanTask> fileScanTaskIterable = TableScanUtil.splitFiles(
+                    scan.planFiles(), scan.targetSplitSize());
+            CloseableIterator<FileScanTask> fileScanTaskIterator = fileScanTaskIterable.iterator();
+            Iterator<FileScanTask> fileScanTasks;
+            boolean canPruneManifests = limit != -1 && !table.isV2Format() && onlyHasPartitionPredicate(table, predicate)
+                    && limit < Integer.MAX_VALUE && enablePruneManifest();
+
+            if (canPruneManifests) {
+                // After iceberg uses partition predicate plan files, each manifests entry must have at least one row of data.
+                fileScanTasks = Iterators.limit(fileScanTaskIterator, (int) limit);
+            } else {
+                fileScanTasks = fileScanTaskIterator;
+            }
+
+            long totalReadCount = 0;
+            while (fileScanTasks.hasNext()) {
+                FileScanTask scanTask = fileScanTasks.next();
+                builder.add(scanTask);
+                totalReadCount += scanTask.file().recordCount();
```

Comment:
what if we change partition spec from day to hour?

Reply:
it's up to the predicate. iceberg native package will adapt to it.

Reply:
currently if table has partition transform, we can't use this limit to prune manifests. It will maintain its original behavior

Reply:
I have baned this optimization when table has partition column evolution